Flaky io.questdb.cutlass.http.DispatcherWriterQueueTest.testAlterTableAddDisconnect

### Describe the bug

failed on Linux, full log attached

```
2022-06-14T11:06:08.8355171Z 2022-06-14T11:06:08.758188Z I i.q.TestListener >>>> io.questdb.cutlass.http.DispatcherWriterQueueTest.testAlterTableAddDisconnect
2022-06-14T11:06:08.8356160Z 2022-06-14T11:06:08.760589Z I i.q.c.t.t.InputFormatConfiguration loading input format config [resource=/text_loader.json]
2022-06-14T11:06:08.8357149Z 2022-06-14T11:06:08.761999Z I i.q.c.m.EngineMigration upgrading database [version=426]
2022-06-14T11:06:08.8357938Z 2022-06-14T11:06:08.762032Z I i.q.c.m.EngineMigration upgraded tables to 426
2022-06-14T11:06:08.8358782Z 2022-06-14T11:06:08.762128Z A IODispatcher listening on 0.0.0.0:9001 [fd=20 backlog=128]
2022-06-14T11:06:08.8359633Z 2022-06-14T11:06:08.763963Z I i.q.g.FunctionFactoryCache loading functions [test=true]
2022-06-14T11:06:08.8362511Z 2022-06-14T11:06:08.768700Z I i.q.g.FunctionFactoryCache loading functions [test=true]
2022-06-14T11:06:08.8363537Z 2022-06-14T11:06:08.773324Z I i.q.g.FunctionFactoryCache loading functions [test=true]
2022-06-14T11:06:08.8364325Z 2022-06-14T11:06:08.776264Z I i.q.c.h.HttpQueryTestBuilder started
2022-06-14T11:06:08.8365118Z 2022-06-14T11:06:08.777910Z I i.q.g.FunctionFactoryCache loading functions [test=true]
2022-06-14T11:06:08.8365899Z 2022-06-14T11:06:08.781231Z I i.q.c.CairoEngine locked [table=`x`, thread=1]
2022-06-14T11:06:08.8366749Z 2022-06-14T11:06:08.782159Z I i.q.c.h.HttpQueryTestBuilder os scheduled [name=questdb-worker-56]
2022-06-14T11:06:08.8367526Z 2022-06-14T11:06:08.783424Z I i.q.c.TableWriter open 'x'
2022-06-14T11:06:08.8368381Z 2022-06-14T11:06:08.783832Z I i.q.c.TableWriter switched partition [path='/tmp/junit1896247758824655740/x/default']
2022-06-14T11:06:08.8369220Z 2022-06-14T11:06:08.784378Z I i.q.c.CairoEngine unlocked [table=`x`]
2022-06-14T11:06:08.8369975Z 2022-06-14T11:06:08.784387Z I i.q.c.p.WriterPool >> [table=`x`, thread=1]
2022-06-14T11:06:08.8370744Z 2022-06-14T11:06:08.792886Z I IODispatcher connected [ip=127.0.0.1, fd=35]
2022-06-14T11:06:08.8371616Z 2022-06-14T11:06:08.793065Z I i.q.c.h.p.JsonQueryProcessorState [35] exec [q='alter table x alter column s add index']
2022-06-14T11:06:08.8372596Z 2022-06-14T11:06:08.793082Z I i.q.c.h.p.QueryCache miss [thread=questdb-worker-56, sql=alter table x alter column s add index]
2022-06-14T11:06:08.8373463Z 2022-06-14T11:06:08.793136Z I i.q.c.p.ReaderPool open 'x' [at=0:0]
2022-06-14T11:06:08.8374325Z 2022-06-14T11:06:08.793466Z I i.q.c.p.WriterPool busy [table=`x`, owner=1, thread=238, reason=test lock]
2022-06-14T11:06:08.8375300Z 2022-06-14T11:06:08.793523Z I i.q.g.e.o.OperationFutureImpl published ASYNC writer command [name=ALTER TABLE,tableName=x,instance=1]
2022-06-14T11:06:08.8376263Z 2022-06-14T11:06:08.793538Z I i.q.c.h.p.JsonQueryProcessor [fd=35] Resource busy, will retry
2022-06-14T11:06:08.8377210Z 2022-06-14T11:06:08.794243Z I i.q.c.TableWriter received async cmd [type=2, tableName=x, tableId=1, correlationId=1, cursor=0]
2022-06-14T11:06:08.8378179Z 2022-06-14T11:06:08.794265Z I i.q.c.TableWriter adding index to 's[SYMBOL, path=/tmp/junit1896247758824655740/x]
2022-06-14T11:06:08.8379043Z 2022-06-14T11:06:08.795021Z I i.q.c.h.HttpConnectionContext retrying query [fd=35]
2022-06-14T11:06:08.8379933Z 2022-06-14T11:06:08.795115Z I i.q.g.e.o.OperationFutureImpl writer command QUERY_STARTED response received [instance=1]
2022-06-14T11:06:08.8380884Z 2022-06-14T11:06:08.795412Z I i.q.c.TableWriter ADDED index to 's[SYMBOL]' to /tmp/junit1896247758824655740/x
2022-06-14T11:06:08.8381873Z 2022-06-14T11:06:08.795423Z I i.q.c.TableWriter published async command complete event [type=2,tableName=x,tableId=1,correlationId=1]
2022-06-14T11:06:08.8382977Z 2022-06-14T11:06:08.795532Z I i.q.c.p.ReaderPool closed 'x' [at=0:0, reason=IDLE]
2022-06-14T11:06:08.8383783Z 2022-06-14T11:06:08.795539Z I i.q.c.p.ReaderPool open 'x' [at=0:0]
2022-06-14T11:06:08.8384546Z 2022-06-14T11:06:08.795764Z I i.q.c.p.WriterPool << [table=`x`, thread=1]
2022-06-14T11:06:08.8385389Z 2022-06-14T11:06:08.795821Z I i.q.c.h.p.JsonQueryProcessorState [35] waiting for update query [instance=1]
2022-06-14T11:06:08.8386260Z 2022-06-14T11:06:08.795831Z I i.q.c.h.p.JsonQueryProcessor [fd=35] Resource busy, will retry
2022-06-14T11:06:08.8387000Z 2022-06-14T11:06:08.795990Z I i.q.c.h.HttpServer closed
2022-06-14T11:06:08.8387701Z 2022-06-14T11:06:08.795995Z I i.q.c.h.p.QueryCache closed
2022-06-14T11:06:08.8388565Z 2022-06-14T11:06:08.796004Z I i.q.c.h.HttpQueryTestBuilder cleaned [worker=0]
2022-06-14T11:06:08.8389280Z 2022-06-14T11:06:08.796062Z I IODispatcher closed
2022-06-14T11:06:08.8389968Z 2022-06-14T11:06:08.796417Z I i.q.c.TableWriter closed 'x'
2022-06-14T11:06:08.8390762Z 2022-06-14T11:06:08.796420Z I i.q.c.p.WriterPool closed [table=`x`, reason=POOL_CLOSED, by=1]
2022-06-14T11:06:08.8391502Z 2022-06-14T11:06:08.796432Z I i.q.c.p.WriterPool closed
2022-06-14T11:06:08.8392296Z 2022-06-14T11:06:08.796517Z I i.q.c.p.ReaderPool closed 'x' [at=0:0, reason=POOL_CLOSED]
2022-06-14T11:06:08.8393070Z 2022-06-14T11:06:08.796521Z I i.q.c.p.ReaderPool closed
2022-06-14T11:06:08.8394018Z 2022-06-14T11:06:08.803335Z E i.q.TestListener ***** Test Failed *****io.questdb.cutlass.http.DispatcherWriterQueueTest.testAlterTableAddDisconnect : 
2022-06-14T11:06:08.8394745Z java.lang.AssertionError: file descriptors [18,35] expected:<1> but was:<2>
2022-06-14T11:06:08.8395257Z 	at org.junit.Assert.fail(Assert.java:89)
2022-06-14T11:06:08.8395756Z 	at org.junit.Assert.failNotEquals(Assert.java:835)
2022-06-14T11:06:08.8396258Z 	at org.junit.Assert.assertEquals(Assert.java:647)
2022-06-14T11:06:08.8396834Z 	at io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:451)
2022-06-14T11:06:08.8397472Z 	at io.questdb.cutlass.http.HttpQueryTestBuilder.run(HttpQueryTestBuilder.java:77)
2022-06-14T11:06:08.8398131Z 	at io.questdb.cutlass.http.HttpQueryTestBuilder.run(HttpQueryTestBuilder.java:73)
2022-06-14T11:06:08.8398851Z 	at io.questdb.cutlass.http.DispatcherWriterQueueTest.runAlterOnBusyTable(DispatcherWriterQueueTest.java:554)
2022-06-14T11:06:08.8399652Z 	at io.questdb.cutlass.http.DispatcherWriterQueueTest.testAlterTableAddDisconnect(DispatcherWriterQueueTest.java:126)
2022-06-14T11:06:08.8520799Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-06-14T11:06:08.8521830Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-06-14T11:06:08.8522592Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-06-14T11:06:08.8523238Z 	at java.lang.reflect.Method.invoke(Method.java:566)
2022-06-14T11:06:08.8523847Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-06-14T11:06:08.8524541Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-06-14T11:06:08.8525223Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-06-14T11:06:08.8525896Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-06-14T11:06:08.8526530Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2022-06-14T11:06:08.8527140Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-06-14T11:06:08.8527783Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-06-14T11:06:08.8528421Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-06-14T11:06:08.8529055Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-06-14T11:06:08.8529732Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-06-14T11:06:08.8530532Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-06-14T11:06:08.8531105Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-06-14T11:06:08.8531704Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-06-14T11:06:08.8532303Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-06-14T11:06:08.8532895Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-06-14T11:06:08.8533491Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-06-14T11:06:08.8534073Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-06-14T11:06:08.8534687Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
2022-06-14T11:06:08.8535454Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
2022-06-14T11:06:08.8536152Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
2022-06-14T11:06:08.8536872Z 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
2022-06-14T11:06:08.8537607Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
2022-06-14T11:06:08.8538280Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2022-06-14T11:06:08.8539460Z 2022-06-14T11:06:08.803466Z I i.q.TestListener <<<< io.questdb.cutlass.http.DispatcherWriterQueueTest.testAlterTableAddDisconnect
```
[56.txt.zip](https://github.com/questdb/questdb/files/8899234/56.txt.zip)



### To reproduce

_No response_

### Expected Behavior

_No response_

### Environment

```markdown
- **QuestDB version**: 6.4.2-SNAPSHOT
- **OS**: Linux
- **Browser**:
```


### Additional context

_No response_

--- a/core/src/main/java/io/questdb/cutlass/http/WaitProcessor.java
+++ b/core/src/main/java/io/questdb/cutlass/http/WaitProcessor.java
@@ -67,6 +67,7 @@ public class WaitProcessor extends SynchronizedJob implements RescheduleContext,
 
     @Override
     public void close() {
+        processInQueue(); // Process incoming queue to close all contexts
         for (int i = 0, n = nextRerun.size(); i < n; i++) {
             Misc.free(nextRerun.poll());
         }

